### ISSUE TITLE
Crashed on Qt 5.13

### ISSUE THREAD

**Issue by NichtsVonChaos** [PhotoFlare/photoflare#246]
Compiled on ubuntu 19.04 by Qt 5.13.1, using GCC 9.
Just run, open a PNG file, then click any point on the picture, it crashed.

Qt reported:
```
QMetaObject::connectSlotsByName: No matching signal for on_image_filtered(QImage)
QMetaObject::connectSlotsByName: No matching signal for on_TextTool_finished()
QMetaObject::connectSlotsByName: No matching signal for on_show_hotspots_triggered()
QDockWidgetLayout::addItem(): please use QDockWidgetLayout::setWidget()
ASSERT: "!isEmpty()" in file /opt/Qt5.13.1/5.13.1/gcc_64/include/QtCore/qvector.h, line 245
```
The stack:
```
    1  __GI_raise                                                                                                                       raise.c            50 
    2  __GI_abort                                                                                                                       abort.c            79 
    3  QMessageLogger::fatal(const char *, ...) const
    4  qt_assert(const char *, const char *, int)
->5  QVector<QPoint>::last                                                                                                            qvector.h          245 
    6  PaintWidget::onSelectionChanged                                                                                                  PaintWidget.cpp    393
    7  QtPrivate::FunctorCall<QtPrivate::IndexesList<0>, QtPrivate::List<QPolygon>, void, void (PaintWidget:: *)(QPolygon)>::call       qobjectdefs_impl.h 152
    8  QtPrivate::FunctionPointer<void (PaintWidget:: *)(QPolygon)>::call<QtPrivate::List<QPolygon>, void>                              qobjectdefs_impl.h 185
    9  QtPrivate::QSlotObject<void (PaintWidget:: *)(QPolygon), QtPrivate::List<QPolygon>, void>::impl                                  qobjectdefs_impl.h 414
    10 QMetaObject::activate(QObject *, int, int, void * *)
    11 Tool::selectionChanged                                                                                                           moc_Tool.cpp       179
    12 PointerTool::onMouseRelease                                                                                                      PointerTool.cpp
    13 PaintWidgetPrivate::mouseReleaseEvent                                                                                            PaintWidget.cpp 
    14 QGraphicsScene::event(QEvent *)
    15 QApplicationPrivate::notify_helper(QObject *, QEvent *)
    16 QApplication::notify(QObject *, QEvent *)
    17 QCoreApplication::notifyInternal2(QObject *, QEvent *)
    18 QGraphicsView::mouseReleaseEvent(QMouseEvent *)
    19 QWidget::event(QEvent *)
    20 QFrame::event(QEvent *)
    21 QGraphicsView::viewportEvent(QEvent *)
    22 QCoreApplicationPrivate::sendThroughObjectEventFilters(QObject *, QEvent *)
    23 QApplicationPrivate::notify_helper(QObject *, QEvent *)
    24 QApplication::notify(QObject *, QEvent *)
    25 QCoreApplication::notifyInternal2(QObject *, QEvent *)
    26 QApplicationPrivate::sendMouseEvent(QWidget *, QMouseEvent *, QWidget *, QWidget *, QWidget * *, QPointer<QWidget>&, bool, bool) 
    27 QWidgetWindow::handleMouseEvent(QMouseEvent *) 
    28 QWidgetWindow::event(QEvent *)
    29 QApplicationPrivate::notify_helper(QObject *, QEvent *) 
    30 QApplication::notify(QObject *, QEvent *) 
    31 QCoreApplication::notifyInternal2(QObject *, QEvent *)
    32 QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent *)
    33 QGuiApplicationPrivate::processWindowSystemEvent(QWindowSystemInterfacePrivate::WindowSystemEvent *) 
    34 QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>)
    35 xcbSourceDispatch(_GSource *, int ( *)(void *), void *)
    36 g_main_context_dispatch 
    37 ?? 
    38 g_main_context_iteration 
    39 QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) 
    40 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>)  
    41 QCoreApplication::exec()  
    42 main                                                                                                                             main.cpp           124
```

**Comment by DylanC** [comment]
@NichtsVonChaos - Thanks for this report. I will look into this as soon as possible.

**Comment by DylanC** [comment]
@NichtsVonChaos - Confirmed issue on Ubuntu 19.04.

**Comment by DylanC** [comment]
@NichtsVonChaos - I can also reproduce on Qt 5.12.4. (Ubuntu 19.10)

**Comment by DylanC** [comment]
@NichtsVonChaos - Fixed via commit 19343db8d48ea1591ce45ab2e0db928dffcca4f3. Thanks!
